### DISCUSSION TITLE
Support for annotations

### DISCUSSION THREAD

**Discussion by risoms** [marcusolsson/grafana-gantt-panel#75] [Ideas]
Thank you for a fantastic plugin! I wanted to see if there's interest in supporting annotations or subsetting each row to include specific events? This would look similar to the nested rows (1, 1.1, 1.2) or markers (D, E, O) included in the image here. 

![image](https://user-images.githubusercontent.com/10383917/139601251-e0e52da6-7cbf-4000-a160-012da8833b27.png)

**Comment by marcusolsson** [comment]
I think it's feature that makes total sense. Though I'm not sure how you'd configure it.

In your case, how would you want to build the query for this?

**Comment by risoms** [comment]
Table-like data that is nested can support this quite easily. For example:

time             | item | subitem | endtime
2021-01-01 | 1      | 1             | 2021-01-10
2021-01-01 | 1      | 2             | 2021-01-04
2021-01-04 | 1      | 3            | 2021-01-07
2021-01-03 | 1      | 4            | 2021-01-10

**Comment by marcusolsson** [comment]
Oh sorry, I was thinking about the markers. I've been thinking about adding support for Grafana annotations, I imagine they would look more like what you see in the graph panel, i.e. vertical bars that cut through all the tasks. Would that solve it?

As for the nested items, what is that you want to do that would require a nested relationship? Is it to be able to collapse groups of tasks, for example?

**Comment by risoms** [comment]
Ah apologies. The title is misleading. I do actually have interest in both. However, I can see markers easier to introduce first. Out of curiosity, do you have a milestone for the support of markers?

**Comment by risoms** [comment]
> Oh sorry, I was thinking about the markers. I've been thinking about adding support for Grafana annotations, I imagine they would look more like what you see in the graph panel, i.e. vertical bars that cut through all the tasks. Would that solve it?

@marcusolsson This is exactly what I'm hoping for, especially if there is a hover tooltip, similar to the timeseries panel.

 
> As for the nested items, what is that you want to do that would require a nested relationship? Is it to be able to collapse groups of tasks, for example? 

Collapsible nested items would be a nice feature, but I think as long as visually it's clear which is the parent and nested items this would suffice.

**Comment by cosmos1978** [comment]
Yes support for annotations would be really helpful.

Its a very nice plugin but not being able to place grafana annotations is a real shortcoming.

**Comment by marcusolsson** [comment]
I've decided to move feature requests to GitHub Discussions. In contrast to bug reports, many feature requests turn into open-ended discussions as it can sometimes be difficult to determine whether a feature is worth implementing. This leads to an increasing number of issues, many which may not be worked on any time soon.

To improve the signal-to-noise ratio among issues, I'm therefore moving all feature requests to the Ideas category under GitHub Discussions. Only when someone's started working on a feature, an actual issue will be created for the feature.

I'll admit that this won't necessarily improve response times, or even guarantee a timely response. Hopefully though, it'll give me a chance to actually clear out issues. Thank you for your patience! 🙏
